### PR TITLE
doc: describe how to update the googleapis SHA

### PR DESCRIPTION
**Pull request by coryan**
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/9008)

### CODE REVIEW

File: doc/contributor/howto-guide-update-googleapis-sha.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
+# How-to Guide: Update googleapis SHA
+
+This document describes the steps required to update the commit SHA for the
+[googleapis][googleapis-repo] repository. The document is intended for
+contributors to the `google-cloud-cpp` libraries, it assumes you are familiar
```
Comment (devbww) [inline doc/contributor/howto-guide-update-googleapis-sha.md:5]: s/libraries, it/libraries. It/

File: doc/contributor/howto-guide-update-googleapis-sha.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+git checkout -b chore-update-googleapis-sha-circa-$(date +%Y-%m-%d)
+```
+
+## Find out the commit SHA you want to use
+
+Use your browser to find out a recent commit SHA for `googleapis`.  In this
+document we will use `134d8be4b58b2bfc373374f71ec52a0df59af664` as an example.
+That was a recent SHA circa 2022-05-18.
+
+Find out what is the SHA256 of the repository tarball at that commit:
```
Comment (devbww) [inline doc/contributor/howto-guide-update-googleapis-sha.md:36]: s/Find out what is the/Find the/

File: doc/contributor/howto-guide-update-googleapis-sha.md [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+bazel build //google/cloud/...
+# CMake caches the old protos, clear them before compiling
+rm -fr build-out/*/cmake-out
+ci/cloudbuild/build.sh -t cmake-install-pr
+```
+
+## Commit the generated changes
+
+```shell
+git commit -m"Re-generate and reformat libraries"
```
Comment (devbww) [inline doc/contributor/howto-guide-update-googleapis-sha.md:82]: s/Re-generate/Regenerate/?

**Review by devbww — APPROVED** [review]
LGTM.

I had investigated whether it was within the powers of Renovate to run `generate-libraries-pr` before sending its PR, and it did not look like such a thing was possible.

Does this fix #7372?  Is there anything left for the KR?

File: doc/contributor/howto-guide-update-googleapis-sha.md (hunk 1 above)
Comment (coryan) [reply doc/contributor/howto-guide-update-googleapis-sha.md]: Fixed.

File: doc/contributor/howto-guide-update-googleapis-sha.md (hunk 2 above)
Comment (coryan) [reply doc/contributor/howto-guide-update-googleapis-sha.md]: Fixed.

File: doc/contributor/howto-guide-update-googleapis-sha.md [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+bazel build //google/cloud/...
+# CMake caches the old protos, clear them before compiling
+rm -fr build-out/*/cmake-out
+ci/cloudbuild/build.sh -t cmake-install-pr
+```
+
+## Commit the generated changes
+
+```shell
+git commit -m"Regenerate and reformat libraries"
```
Comment (devbww) [inline doc/contributor/howto-guide-update-googleapis-sha.md:82]: Opt: Perhaps "and reformat" is overkill?

Comment (coryan) [reply doc/contributor/howto-guide-update-googleapis-sha.md]: Done.  I was also thinking that `git commit --amend --no-edit` would make it easier to create a PR, as the PR title would match the only commit.  Thoughts?

**Pull request merged**